CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required( VERSION 2.8 )
+
+project(plotIt LANGUAGES CXX)
+
+include(GNUInstallDirs)
+include(ExternalProject)
+
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
+
+if(EXISTS "$ENV{CMAKE_PREFIX_PATH}/include/boost")
+    message(STATUS "Will use $ENV{CMAKE_PREFIX_PATH} as base path for boost")
+    set(BOOST_ROOT $ENV{CMAKE_PREFIX_PATH})
+    set(Boost_NO_BOOST_CMAKE ON)
+endif()
+
+find_package( ROOT REQUIRED )
+find_package( Boost REQUIRED COMPONENTS filesystem regex system )
+
+message(STATUS "Found Boost with include dir ${Boost_INCLUDE_DIR} (change by passing e.g. '-DBoost_NO_BOOST_CMAKE=ON -DBOOST_ROOT=$(scram tool tag boost BOOST_BASE)')")
+
+ExternalProject_Add(
+  yaml-cpp
+  URL https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.6.2.tar.gz
+  CMAKE_ARGS -DYAML_CPP_BUILD_TOOLS=OFF -DYAML_CPP_BUILD_CONTRIB=OFF -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/external
+  )
+ExternalProject_Add(
+  tclap
+  URL https://github.com/eile/tclap/archive/tclap-1-2-1-release-final.tar.gz
+  BUILD_IN_SOURCE 1
+  PATCH_COMMAND ./autotools.sh
+  CONFIGURE_COMMAND ./configure --prefix=${CMAKE_CURRENT_BINARY_DIR}/external
+  )
+
+set(SRCS
+  src/plotIt.cc
+  src/summary.cc
+  src/systematics.cc
+  src/TH1Plotter.cc
+  src/types.cc
+  src/utilities.cc
+  src/uuid.cc
+  )
+
+add_executable(plotIt ${SRCS})
+add_dependencies(plotIt yaml-cpp tclap)
+target_compile_features(plotIt PRIVATE cxx_std_17)
```

Comment:
do we really need C++17? this won't work with older gcc versions, no?

Reply:
you're right... copy-pasted from somewhere else ;-). We can let cmake choose (the user can pass command-line options to it to choose anyway)

Reply:
so this didn't work by just removing it, because recent ROOT builds have C++17, also in the interfaces, so to get it right without forcing, I should depend on the targets exported by ROOT's cmake (which are not in my simple find module).
Since the goal of this was to support recent versions, I did exactly that: removed the find module, and depend on the specific Boost and ROOT targets that are needed (for compatibility with older platforms there's still the Makefile - replicating all this correctly for those cases would be quite a bit of work).

Reply:
Yeah I remember that's a pain... I'm fine with it only working with recent versions since there's the Makefile as a fallback.